### PR TITLE
Implemented gossip menu for npc 2239 with english and german locales

### PR DESCRIPTION
**Pull request by schell244**
fix for: https://github.com/vmangos/core/issues/356

### CODE REVIEW

File: sql/migrations/20200129094000_world.sql [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+SET v = (SELECT COUNT(*) FROM `migrations` WHERE `id`='20200129094000');
+IF v=0 THEN
+INSERT INTO `migrations` VALUES ('20200129094000');
+-- Add your query below.
+
+-- Added gossip for npc 2239 (Drull) with english & german locales
+update `creature_template` set gossip_menu_id=262 where entry=2239;
+
+-- Greeting
+insert into `gossip_menu` (`entry`, `text_id`, `condition_id`) values('262','2','0');
```
Comment (alexkulya) [inline sql/migrations/20200129094000_world.sql]: text_id = 759

File: sql/migrations/20200129094000_world.sql [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+insert into `npc_text` (`ID`, `BroadcastTextID0`, `Probability0`, `BroadcastTextID1`, `Probability1`, `BroadcastTextID2`, `Probability2`, `BroadcastTextID3`, `Probability3`, `BroadcastTextID4`, `Probability4`, `BroadcastTextID5`, `Probability5`, `BroadcastTextID6`, `Probability6`, `BroadcastTextID7`, `Probability7`) values('2','2','1','0','0','0','0','0','0','0','0','0','0','0','0','0','0');
```
Comment (alexkulya) [inline sql/migrations/20200129094000_world.sql]: npc_text ID: 759 (already contains the necessary BCT: 2742)

File: sql/migrations/20200129094000_world.sql [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+insert into `locales_broadcast_text` (`ID`, `MaleText_loc1`, `MaleText_loc2`, `MaleText_loc3`, `MaleText_loc4`, `MaleText_loc5`, `MaleText_loc6`, `MaleText_loc7`, `MaleText_loc8`, `FemaleText_loc1`, `FemaleText_loc2`, `FemaleText_loc3`, `FemaleText_loc4`, `FemaleText_loc5`, `FemaleText_loc6`, `FemaleText_loc7`, `FemaleText_loc8`, `VerifiedBuild`) values('2',NULL,NULL,'Verspottet Ihr mich, während ich nicht in der Lage bin mich zu verteidigen, gebunden an diese Eisenkugel?',NULL,NULL,NULL,NULL,NULL,NULL,NULL,'Verspottet Ihr mich, während ich nicht in der Lage bin mich zu verteidigen, gebunden an diese Eisenkugel?',NULL,NULL,NULL,NULL,NULL,'0');
+
+-- Gossip Option
+insert into `gossip_menu_option` (`menu_id`, `id`, `option_icon`, `option_text`, `OptionBroadcastTextID`, `option_id`, `npc_option_npcflag`, `action_menu_id`, `action_poi_id`, `action_script_id`, `box_coded`, `box_money`, `box_text`, `BoxBroadcastTextID`, `condition_id`) values('262','0','0','How can I free you?','8','1','3','263','0','0','0','0',NULL,'0','0');
```
Comment (alexkulya) [inline sql/migrations/20200129094000_world.sql]: BCT: 2746

File: sql/migrations/20200129094000_world.sql [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+insert into `broadcast_text` (`ID`, `MaleText`, `FemaleText`, `Sound`, `Type`, `Language`, `EmoteId0`, `EmoteId1`, `EmoteId2`, `EmoteDelay0`, `EmoteDelay1`, `EmoteDelay2`) values('8','How can I free you?','How can I free you?','0','0','0','0','0','0','0','0','0');
+insert into `locales_broadcast_text` (`ID`, `MaleText_loc1`, `MaleText_loc2`, `MaleText_loc3`, `MaleText_loc4`, `MaleText_loc5`, `MaleText_loc6`, `MaleText_loc7`, `MaleText_loc8`, `FemaleText_loc1`, `FemaleText_loc2`, `FemaleText_loc3`, `FemaleText_loc4`, `FemaleText_loc5`, `FemaleText_loc6`, `FemaleText_loc7`, `FemaleText_loc8`, `VerifiedBuild`) values('8',NULL,NULL,'Wie kann ich Euch befreien?',NULL,NULL,NULL,NULL,NULL,NULL,NULL,'Wie kann ich Euch befreien?',NULL,NULL,NULL,NULL,NULL,'0');
+
+-- Gossip Option Page
+insert into `gossip_menu` (`entry`, `text_id`, `condition_id`) values('263','3','0');
```
Comment (alexkulya) [inline sql/migrations/20200129094000_world.sql]: text_id = 760

File: sql/migrations/20200129094000_world.sql [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+insert into `npc_text` (`ID`, `BroadcastTextID0`, `Probability0`, `BroadcastTextID1`, `Probability1`, `BroadcastTextID2`, `Probability2`, `BroadcastTextID3`, `Probability3`, `BroadcastTextID4`, `Probability4`, `BroadcastTextID5`, `Probability5`, `BroadcastTextID6`, `Probability6`, `BroadcastTextID7`, `Probability7`) values('3','20','1','0','0','0','0','0','0','0','0','0','0','0','0','0','0');
```
Comment (alexkulya) [inline sql/migrations/20200129094000_world.sql]: npc_text ID: 760 (already contains the necessary BCT: 2744)

**Comment by schell244** [comment]
thanks @alexkulya

**Comment by ratkosrb** [comment]
Never insert into `broadcast_text`. That whole table has been sniffed from classic and it has all the texts that existed in vanilla. If something cannot be found, then it was added in a later expansion, or its completely custom. Also `gossip_menu` and `npc_text` ids are sniffed too, and you should not make up your own ids. Check other projects like cmangos and trinity if you are trying to find out the real ids.

**Comment by schell244** [comment]
That's good to know - it was the first time for me working with texts/gossips. For further issues, I know where to look then.

**Pull request merged**